File: VB/WindowsFormsApplication1/Form1.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraPivotGrid
Imports System
Imports System.Data

Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			excelDataSource1.FileName = "SalesPerson.xlsx"
			excelDataSource1.Fill()
			pivotGridControl1.BestFit()
			pivotGridControl1.BestFitDataHeaders(True)

		End Sub

		Private Sub pivotGridControl1_CellDoubleClick(ByVal sender As Object, ByVal e As PivotCellEventArgs) Handles pivotGridControl1.CellDoubleClick
			If e.RowField Is Nothing Then
				Return 'Grand Total cell
			End If
			Dim columnFields() As PivotGridField = e.GetColumnFields()
			Dim columnValues(columnFields.Length - 1) As Object
			For i As Integer = 0 To columnFields.Length - 1
				columnValues(i) = e.GetFieldValue(columnFields(i))
			Next i
			Dim rowFields() As PivotGridField = e.GetRowFields()
			Dim rowValues(rowFields.Length - 1) As Object
			For i As Integer = 0 To rowFields.Length - 2
				rowValues(i) = e.GetFieldValue(rowFields(i))
			Next i

			Dim dt As New DataTable()
			For Each field As PivotGridField In columnFields
				dt.Columns.Add(field.ToString(), field.DataType)
			Next field
			For Each field As PivotGridField In rowFields
				dt.Columns.Add(field.ToString(), field.DataType)
			Next field
			dt.Columns.Add(e.DataField.ToString(), e.DataField.DataType)

			For Each lastLevelValue As Object In e.RowField.GetVisibleValues()
				rowValues(rowFields.Length-1) = lastLevelValue
				Dim cellValue As Object = e.GetCellValue(columnValues, rowValues, e.DataField)
				If cellValue Is Nothing Then
					Continue For
				End If
				Dim row As DataRow = dt.NewRow()

				For i As Integer = 0 To columnFields.Length - 1
					row(columnFields(i).ToString()) = columnValues(i)
				Next i
				For i As Integer = 0 To rowFields.Length - 1
					row(rowFields(i).ToString()) = rowValues(i)
				Next i
				row(e.DataField.ToString()) = cellValue
				dt.Rows.Add(row)

			Next lastLevelValue
			gridControl1.DataSource = dt
			gridView1.BestFitColumns()
		End Sub
	End Class
End Namespace

File: CS/WindowsFormsApplication1/Form1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraPivotGrid;
using System;
using System.Data;

namespace WindowsFormsApplication1
{
    public partial class Form1 : XtraForm
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.invoicesTableAdapter.Fill(this.nwindDataSet.Invoices);
            pivotGridControl1.BestFit();
            pivotGridControl1.BestFitColumnArea();

        }

        private void pivotGridControl1_CellDoubleClick(object sender, DevExpress.XtraPivotGrid.PivotCellEventArgs e)
        {
            if (e.RowField == null) return; //Grand Total cell
            PivotGridField[] columnFields = e.GetColumnFields();
            object[] columnValues = new object[columnFields.Length];
            for (int i = 0; i < columnFields.Length; i++)
                columnValues[i] = e.GetFieldValue(columnFields[i]);
            PivotGridField[] rowFields = e.GetRowFields();
            object[] rowValues = new object[rowFields.Length ];
            for (int i = 0; i < rowFields.Length -1; i++)
                rowValues[i] = e.GetFieldValue(rowFields[i]);

            DataTable dt = new DataTable();
            foreach (PivotGridField field in columnFields)
                dt.Columns.Add(field.ToString(), field.DataType);
            foreach (PivotGridField field in rowFields)
                dt.Columns.Add(field.ToString(), field.DataType);
            dt.Columns.Add(e.DataField.ToString(), e.DataField.DataType);

            foreach (object lastLevelValue in e.RowField.GetVisibleValues() )
            {
                rowValues[rowFields.Length-1] = lastLevelValue;
                object cellValue = e.GetCellValue(columnValues, rowValues, e.DataField);
                if (cellValue == null) continue;
                DataRow row = dt.NewRow();

                for (int i = 0; i < columnFields.Length; i++)
                    row[columnFields[i].ToString()] = columnValues[i];
                for (int i = 0; i < rowFields.Length; i++)
                    row[rowFields[i].ToString()] = rowValues[i];
                row[e.DataField.ToString()] = cellValue;
                dt.Rows.Add(row);
                
            }
            dataGridView1.DataSource = dt;
            dataGridView1.AutoResizeColumns();
        }
    }
}
